crates/miden-tx/src/host/mod.rs

```diff
@@ -481,26 +488,13 @@ impl<A: AdviceProvider> Host for TransactionHost<A> {
     }
 
     fn on_event(&mut self, process: ProcessState, event_id: u32) -> Result<(), ExecutionError> {
-        // Handle only the FalconSigToStack event here. All other SystemEvents should be handled by
-        // the VM.
-        match SystemEvent::from_event_id(event_id) {
-            Some(SystemEvent::FalconSigToStack) => {
-                return self.on_signature_requested(process);
-            },
-            Some(_) => {
-                return Err(ExecutionError::EventError(Box::new(
-                    TransactionEventError::InvalidTransactionEvent(event_id),
-                )));
-            },
-            // If the event is not a SystemEvent, continue and try parsing it as a
-            // TransactionEvent.
-            None => (),
-        }
-
         let transaction_event = TransactionEvent::try_from(event_id)
             .map_err(|err| ExecutionError::EventError(Box::new(err)))?;
 
-        if process.ctx() != ContextId::root() {
+        // only the `FalconSigToStack` event can be executed outside the root context
+        if process.ctx() != ContextId::root()
+            && !matches!(transaction_event, TransactionEvent::FalconSigToStack)
```

Comment:
@bobbinth not sure what is the cleanest way to do this

Reply:
I think this is fine for now.